adapters/repos/db/nodes.go

```diff
@@ -55,46 +56,78 @@ func (db *DB) getNodeStatus(ctx context.Context, nodeName string) (*models.NodeS
 }
 
 // IncomingGetNodeStatus returns the index if it exists or nil if it doesn't
-func (db *DB) IncomingGetNodeStatus(ctx context.Context) (*models.NodeStatus, error) {
-	return db.localNodeStatus(), nil
+func (db *DB) IncomingGetNodeStatus(ctx context.Context, className string) (*models.NodeStatus, error) {
+	return db.localNodeStatus(className), nil
 }
 
-func (db *DB) localNodeStatus() *models.NodeStatus {
-	var totalObjectCount int64
-	var shardCount int64
-	shards := []*models.NodeShardStatus{}
-	db.indexLock.RLock()
-	for _, index := range db.indices {
-		index.ForEachShard(func(name string, shard *Shard) error {
-			objectCount := int64(shard.objectCount())
-			shardStatus := &models.NodeShardStatus{
-				Name:        name,
-				Class:       shard.index.Config.ClassName.String(),
-				ObjectCount: objectCount,
-			}
-			totalObjectCount += objectCount
-			shardCount++
-			shards = append(shards, shardStatus)
-			return nil
-		})
+func (db *DB) localNodeStatus(className string) *models.NodeStatus {
+	var (
+		objectCount int64
+		shards      []*models.NodeShardStatus
+	)
+
+	if className == "" {
+		objectCount = db.localNodeStatusAll(&shards)
+	} else {
+		objectCount = db.localNodeStatusForClass(&shards, className)
 	}
-	db.indexLock.RUnlock()
 
 	clusterHealthStatus := models.NodeStatusStatusHEALTHY
 	if db.schemaGetter.ClusterHealthScore() > 0 {
 		clusterHealthStatus = models.NodeStatusStatusUNHEALTHY
 	}
 
-	status := &models.NodeStatus{
+	return &models.NodeStatus{
 		Name:    db.schemaGetter.NodeName(),
 		Version: db.config.ServerVersion,
 		GitHash: db.config.GitHash,
 		Status:  &clusterHealthStatus,
 		Shards:  shards,
 		Stats: &models.NodeStats{
-			ShardCount:  shardCount,
-			ObjectCount: totalObjectCount,
+			ShardCount:  int64(len(shards)),
+			ObjectCount: objectCount,
 		},
 	}
-	return status
+}
+
+func (db *DB) localNodeStatusAll(status *[]*models.NodeShardStatus) (totalCount int64) {
+	db.indexLock.RLock()
+	for name, idx := range db.indices {
+		if idx == nil {
+			db.logger.WithField("action", "local_node_status_for_all").
+				Warningf("no resource found for index %q", name)
+			continue
+		}
+		totalCount += idx.getShardsNodeStatus(status)
+	}
+	db.indexLock.RUnlock()
+	return
+}
+
+func (db *DB) localNodeStatusForClass(status *[]*models.NodeShardStatus,
+	className string,
+) (totalCount int64) {
+	idx := db.GetIndex(schema.ClassName(className))
+	if idx == nil {
+		db.logger.WithField("action", "local_node_status_for_class").
+			Warningf("no index found for class %q", className)
+		// Fallback to returning all classes
+		return db.localNodeStatusAll(status)
```

Comment:
I would rather see empty result than all classes combined, since I specifically requested for particular class

Reply:
agreed, fixed